src/vvwe.rs

```diff
@@ -0,0 +1,297 @@
+// Causality barrier
+// Keeps for each known peer, keeps track of the latest clock seen
+// And a set of messages that are from the future
+// and outputs the full-in-order sequence of messages
+//
+//
+
+#![allow(missing_docs)]
+
+use derive_more::{Add, From, Into};
+use std::collections::*;
+use serde::{Deserialize, Deserializer, self, Serialize};
+
+/// Temporary
+#[derive(PartialEq, Eq, Debug, Copy, Clone, Hash, PartialOrd, Ord, From, Deserialize, Serialize)]
+pub struct SiteId(
+    #[serde(deserialize_with = "deserialize_number_from_string")]
```

Comment:
Why is this needed? shouldn't serde just deserialize it the right way?

Reply:
https://github.com/serde-rs/json/issues/317

I need to check if it's still required here but the problem I was having was that the number would get turned into a stringified number by serde / js and then break deserialization :/

Reply:
So it's some sort of compatibility code? For the case that the value was serialized as as string?

Reply:
Yea, it's been a while since I originally wrote that but I recall it having to do with serde generating weird instances for newtyped numbers.

Reply:
I'm planning on getting rid of the siteid entirely in favor of the more generic `Actor` trait